### ISSUE TITLE
Translation proposal

### ISSUE THREAD

**Issue by DenB10** [jeevatkm/ReplyWithHeaderMozilla#150]
Hi jeevatkm,

I find your extension interesting, so I've installed it.
I then had fun translating the options page into my language (French), then implementing localization and finally arranging it to suit the dark theme.
I'm sending you the result below, which you can do what you like with, retain or discard if you're not interested.

[replywithheader-2.6.0-tb.zip](https://github.com/jeevatkm/ReplyWithHeaderMozilla/files/15366334/replywithheader-2.6.0-tb.zip)

Best regards,

DenB

**Comment by jeevatkm** [comment]
@DenB10 Thanks for your appreciation. 

I have put my effort into creating the RWH v3 mail extension currently in v3 beta2. Can you please try this version and share feedback? 

Also, if you could help translate v3, it would benefit all RWH users.

See here https://github.com/jeevatkm/ReplyWithHeaderMozilla/issues/103#issuecomment-2114025131

**Comment by jeevatkm** [comment]
Just so you know, the v2.6.0 version is a legacy model of the TB addon.

**Comment by DenB10** [comment]
Version 2.6.0 is the latest version online on ATN, I believe? That's what I've been tinkering with.
But I'm going to try v3 beta 2...

**Comment by jeevatkm** [comment]
V3.0.0 is an upcoming new version written newly from the ground up. It is currently at the beta2 stage.

**Comment by DenB10** [comment]
Yes, it's a beta but I can't install it because I'm still using Thunderbird version 115 and the minimum beta version is 125.0.

**Comment by jeevatkm** [comment]
You're correct; you must take a few manual steps to achieve it. 

* Manual Installation instructions here - https://support.mozilla.org/en-US/kb/installing-addon-thunderbird
* TB v125 download location - https://ftp.mozilla.org/pub/thunderbird/releases/125.0/

**Comment by DenB10** [comment]
I understand, but I don't actually use Thunderbird. For some reasons, I use Betterbird (www.betterbird.eu), which is an improved fork of Thunderbird, and follow the normal Thunderbird distribution channel, i.e. 115.11.0 currently.
I'm going to try and get a portable version of Thunderbird 125 minimum...

**Comment by jeevatkm** [comment]
@DenB10 I see you're using Betterbird. Sounds good.

**Comment by DenB10** [comment]
Bad news:
I won't be able to help you by testing RWH version 3.
I did get a portable version of Thunderbird 127, but I can't open it. After some research, I discovered that Thunderbird 127 requires at least Windows 10, and as you can guess, I'm still using Windows 7.
Sorry about that.

Good news though:
By lowering the strict minimum version to 115.0 in the RWH 3 manifest, I was able to install it.
Of course, it doesn't work, but I was still able to perform the localization.
Here it is:
[replywithheader-v3.0.0.beta2 (with localization).zip](https://github.com/jeevatkm/ReplyWithHeaderMozilla/files/15369985/replywithheader-v3.0.0.beta2.with.localization.zip)
(it's a zip because Github doesn't accept xpi)
It's up to you to decide what you want to do with it.

**Comment by jeevatkm** [comment]
@DenB10 Thanks for sharing it. I will have a look.

FYI, RWH v3 will not be below TB v125 because TB is missing a few APIs.

**Comment by topoldo** [comment]
Hi **jeevatkm**!
I would like to create the translation into Italian for your add-on.
I saw you did not provide a locale subdir with files supporting msgid/msgstr varibles to be translated.
Do you plan to do this, or is there any workaround for it?
Plese let me know. I candidate myself for the Italian version.
Yours sincerely,
Leopoldo Saggin aka Topoldo

**Comment by jeevatkm** [comment]
@topoldo Thanks for reaching out and volunteering for translation into Italian. 

Yes, there is dev effort required to open up the translation in RWH. That's why I have not yet merged @DenB10’s contribution of the French translation.

I will reach out to you when I wrap up the initial dev effort of translation.

**Comment by DenB10** [comment]
@jeevatkm

Hello, I don't think you've tested the zip file I posted last May. 

It's not just the French translation I'm proposing, but also the localization implementation. I simply added a few lines of code to the options.js file. I've tested it on a Windows 10 virtual machine in which I've installed tb128 (I can't install tb128 directly because my computer is still running Windows 7) and it works fine...

**Comment by jeevatkm** [comment]
@DenB10 While ago I have looked at the content of zip extract. Thanks for implementing localization. I will take your contribution and make required changes. Thanks again.

**Comment by topoldo** [comment]
Herein I attach my translation into Italian for RWH in the .zip format.
It was based on the "locale" section of the beta version of this app as posted in this thread by @DenB10 inside his May19 comment.

Please note:
1. In my translation (and also in the French one, I suppose) there are **some strings** which are **missing** because of some integrations added since last May19 in the official version RWH.
2. In the Italian language, we use several **apostrophes**. Please note that **I DID NOT ESCAPED** them.
However, if RWH needs them,please let me know so I can add them.

Topoldo
[messages_it.zip](https://github.com/user-attachments/files/17976752/messages_it.zip)

**Comment by DenB10** [comment]
@jeevatkm
>While ago I have looked at the content of zip extract. Thanks for implementing localization. I will take your contribution and make required changes. Thanks again.

You're welcome. I've updated the localization for version 3.3.0 and included the Italian translation from @topoldo:
[replywithheader-3.3.0.mod.zip](https://github.com/user-attachments/files/17981248/replywithheader-3.3.0.mod.zip)

@topoldo 
I've taken the liberty of translating into Italian the few additional strings that were not present in version 3.0.0.beta2. But as I don't speak Italian (I used an automatic translator), can you check the translation?

**Comment by topoldo** [comment]
@DenB10 & @jeevatkm
I improved a little bit my previous translation of RWH into Italian and I modified some strings translated by @DenB10.
I herein attach the zip of my latest translation.

[messages_it_new.zip](https://github.com/user-attachments/files/17982123/messages_it_new.zip)

Topoldo

PS: In my Win 11 24H2 with Betterbird 128.5.0esr-bb18 (64 bit) I removed the official RWH-3.3.0 and I substituted with the @DenB10 modified one. I could install it successfully, but I can't access Options. So something is broken.
I checked the syntax of my translation and seems ok.

**Comment by topoldo** [comment]
Correction: now it works (so I can check also the translation ;-)

**Comment by DenB10** [comment]
@topoldo 
Very strange, I just did the same thing as you and no problem. Did you restart Betterbird after uninstalling the “official” RWH and before installing the localized version?

PS: the updated localized version: 
[replywithheader-3.3.0.mod.zip](https://github.com/user-attachments/files/17982700/replywithheader-3.3.0.mod.zip)

**Comment by topoldo** [comment]
I had to make a little modification to my translation.
So herein I post the **definitive** one.
Topoldo

[RWH_messages_it_latest.zip](https://github.com/user-attachments/files/17982718/RWH_messages_it_latest.zip)

**Comment by topoldo** [comment]
Everything now is working :-)

**Comment by DenB10** [comment]
Perfect. Last proposal? -> 
[replywithheader-3.3.0(+localization).zip](https://github.com/user-attachments/files/17983885/replywithheader-3.3.0.%2Blocalization.zip)
We still need to hear from @jeevatkm

**Comment by topoldo** [comment]
@DenB10: the version of RWH you put in your latest message **DOES NOT** contain my **latest** transaltion into Italian, which is:

[RWH_messages_it_latest.zip](https://github.com/user-attachments/files/17982718/RWH_messages_it_latest.zip).

It contains instead my **penultimate** version, ie:
[messages_it_new.zip](https://github.com/user-attachments/files/17982123/messages_it_new.zip)

Could you please integrate the **correct** translation into your _definitive_ proposal RWH zip file?

TIA
Topoldo

**Comment by DenB10** [comment]
I don't get it. Are you sure?
I've just downloaded the file I recently uploaded (replywithheader-3.3.0(+localization).zip) and the  file with your latest translation (RWH_messages_it_latest.zip). I've compared the two translation files several times and have come to the same conclusion: it's your latest translation that's in my zip archive!

**Comment by topoldo** [comment]
@DenB10: You're right!
There is only one difference between the two files of translation I provided.
After the installation of
[replywithheader-3.3.0(+localization).zip](https://github.com/user-attachments/files/17983885/replywithheader-3.3.0.%2Blocalization.zip)
I went immediately to check if that modification was present or not.
Maybe because of the cache of the page, I saw the old string instead of the latest one.

Now I rechecked everything and I can confirm that your file:
[replywithheader-3.3.0(+localization).zip](https://github.com/user-attachments/files/17983885/replywithheader-3.3.0.%2Blocalization.zip)
**is correct** because it contains my latest version of translation

Topoldo

**Comment by jeevatkm** [comment]
Thanks for @DenB10 @topoldo for collaborating together to achieve the translation. Coming weekend, I will take out time to integrate the update from this https://github.com/jeevatkm/ReplyWithHeaderMozilla/issues/150#issuecomment-2513147595 and will create a dev branch. 

After that, I will request your review. If any of the changes are required, you could create a PR against the dev branch for each translation message with corrections/improvements.

**Comment by topoldo** [comment]
It'seems that the update to RWH 3.4.0 is still pending...
I'm using the @DenB10 version which is fine! ,-)
